public_html/wikiWhoProxy.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+header( 'Content-Type: application/json' );
+header( 'Access-Control-Allow-Origin: *' );
+
+// Restrict access to Wikipedias.
+if ( 1 !== preg_match( '/^https:\/\/.*?\.wikipedia\.org/', $_SERVER[ 'HTTP_REFERER' ] ) ) {
+    http_response_code( 403 );
+    die();
+}
+
+$cnf = parse_ini_file( '../wikiwho.ini' );
+if ( !$cnf || !$cnf[ 'user' ] || !$cnf[ 'password' ] ) {
+    throw new Exception(
+        'WikiWho API credentials not found at ' . dirname( __FILE__, 2 ) . '/wikiwho.ini'
+    );
+}
+
+// We strip out /wikiwho for the Toolforge location, and /wikiWhoProxy.php as the entry point when testing locally.
```

Comment:
How are you testing locally? Adding the Referer manually?

Reply:
That was from before I added the referer check. I'll remove that part of the comment.